qiskit/opflow/primitive_ops/tapered_pauli_sum_op.py

```diff
@@ -391,7 +418,8 @@ def _taper(self, op: PauliSumOp, curr_tapering_values: List[int]) -> OperatorBas
             z_temp = np.delete(pauli_term.primitive.paulis.z[0].copy(), np.asarray(self._sq_list))
             x_temp = np.delete(pauli_term.primitive.paulis.x[0].copy(), np.asarray(self._sq_list))
             pauli_list.append((Pauli((z_temp, x_temp)).to_label(), coeff_out))
-        spo = SparsePauliOp.from_list(pauli_list).simplify(atol=0.0)
+
+        spo = SparsePauliOp.from_list(pauli_list).simplify(atol=self.atol, rtol=self.rtol)
```

Comment:
This changes the `atol` value which is being used when tapering. While I think that it makes sense to rely on the operators internal value (defaulting to `TolerancesMixin._ATOL_DEFAULT=1e-8`) we should verify why the value was set to `0.0` to begin with. @ikkoham do you have any input on this?

Reply:
Probably to make it consistent with Aqua

Reply:
This line:
https://github.com/Qiskit/qiskit-aqua/blob/1f2c316c3a1aca1296f45241d14ad8ae5fbe2027/qiskit/aqua/operators/legacy/weighted_pauli_operator.py#L1227

I don't know why this sets 0 as the threshold here in the first place.

Reply:
@woodsp-ibm @pbark Do either of you know why this threshold was explicitly set to `0.0`?

Reply:
This is actually from https://github.com/Qiskit/qiskit-aqua/pull/731 maybe @manoelmarques  remembers why it was necessary 😄

Reply:
Okay that actually makes sense. We have seen in the past that actual-zero values can cause yet other problems so it makes sense that these were being removed.

However, I agree with @woodsp-ibm that truncating at a 1e-8 default is a bit too aggressive (especially when considering this in the context of chemistry applications). In that sense we should probably stick to the old way of removing pure-zero terms (or remove that if you think it's no longer necessary) and let the user (in this case Nature) handle setting a sensible truncation threshold when constructing the `Z2Symmetries`.

Reply:
Changed to 0 tolerance per default in ade48e197